From 3f4a5a1f9935029b791214f7a64d1c6536191c27 Mon Sep 17 00:00:00 2001
From: Daniele Palmas <dnlplm@gmail.com>
Date: Tue, 15 Feb 2022 14:00:29 +0100
Subject: [PATCH 10/12] drivers: bus: mhi: remove unsupported MBIM channels for
 FN980

Signed-off-by: Daniele Palmas <dnlplm@gmail.com>
---
 drivers/bus/mhi/pci_generic.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/drivers/bus/mhi/pci_generic.c b/drivers/bus/mhi/pci_generic.c
index ce137558fa39..7a613992ede0 100644
--- a/drivers/bus/mhi/pci_generic.c
+++ b/drivers/bus/mhi/pci_generic.c
@@ -438,8 +438,6 @@ static const struct mhi_channel_config mhi_telit_fn980_channels[] = {
 	MHI_CHANNEL_CONFIG_DL_SBL(3, "SAHARA", 32, 0),
 	MHI_CHANNEL_CONFIG_AMSS_SBL_UL(4, "DIAG", 64, 1),
 	MHI_CHANNEL_CONFIG_AMSS_SBL_DL(5, "DIAG", 64, 1),
-	MHI_CHANNEL_CONFIG_UL(12, "MBIM", 4, 0),
-	MHI_CHANNEL_CONFIG_DL(13, "MBIM", 4, 0),
 	MHI_CHANNEL_CONFIG_UL(14, "QMI", 16, 0),
 	MHI_CHANNEL_CONFIG_DL(15, "QMI", 16, 0),
 	MHI_CHANNEL_CONFIG_UL(20, "IPCR", 16, 0),
@@ -481,8 +479,6 @@ static const struct mhi_pci_dev_info mhi_telit_fn980_info = {
 };
 
 static const struct mhi_channel_config mhi_telit_fn980_hw_v1_channels[] = {
-	MHI_CHANNEL_CONFIG_UL(12, "MBIM", 4, 0),
-	MHI_CHANNEL_CONFIG_DL(13, "MBIM", 4, 0),
 	MHI_CHANNEL_CONFIG_UL(14, "QMI", 16, 0),
 	MHI_CHANNEL_CONFIG_DL(15, "QMI", 16, 0),
 	MHI_CHANNEL_CONFIG_UL(20, "IPCR", 16, 0),
-- 
2.32.0

