From 12f0ff6eb5464be75d722865a66d910fc8de22d2 Mon Sep 17 00:00:00 2001
From: Daniele Palmas <dnlplm@gmail.com>
Date: Mon, 4 Nov 2024 10:25:44 +0100
Subject: [PATCH 30/31] net: mhi: remove wrong skb queuing in mhi_ndo_xmit

---
 drivers/net/mhi_net.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/net/mhi_net.c b/drivers/net/mhi_net.c
index fb14b52f3ed6..c085a7bb75d2 100644
--- a/drivers/net/mhi_net.c
+++ b/drivers/net/mhi_net.c
@@ -79,7 +79,6 @@ static netdev_tx_t mhi_ndo_xmit(struct sk_buff *skb, struct net_device *ndev)
 	unsigned int queue = 0;
 	struct netdev_queue *txq;
 
-	err = mhi_queue_skb(mdev, DMA_TO_DEVICE, skb, skb->len, MHI_EOT);
 	/* Get the queue mapping number to transfer to the right channel*/
 	if (mhi_netdev->queue)
 		queue = skb_get_queue_mapping(skb);
-- 
2.43.0

